### DISCUSSION TITLE
Comment multiple lines on notepad++

### DISCUSSION THREAD

**Discussion by yoongschoi** [nrc-cnrc/EGSnrc#1033] [Questions] [answered]
I write my code files in .egsinp on notepad++. How can I comment multiple lines at once? It's such time-consuming to put # line by line..

**Comment by ftessier** [comment, marked as answer]
Use a code editor that understands syntax (e.g., VS Code). Select the language Ruby, and you should then be able to comment multiple lines.

Alternatively, you can also comment out only the `#:start` tag to effectively comment out an entire block (until the `:stop` tag), because the input parser looks for `:start` tags to trigger the block input.

**Reply by rtownson** [reply]
@yoonseungchoi  I agree that switching to VS code or similar would be better. But notepad++ does have block commenting, toggled using `CTRL+Q` or `CTRL+SHIFT+Q`, if the editor has been configured to assign a programming language to the filetype (e.g. ruby).

**Reply by yoongschoi** [reply]
@ftessier #:start worked! thank you so much. Can't code editors recognise an .egsinp file as a code file?  because the comment shortcut didn't work for the file.

**Reply by yoongschoi** [reply]
@rtownson Yeah it would have been better indeed cause I anyways use a code editor (I use Spyder) for Python codes at the same time. But Spyder just simply doesn't recognise the .egsinp file as a code file and the comment shortcut didn't work.

**Comment by rtownson** [comment]
> Can't code editors recognise an .egsinp file as a code file? because the comment shortcut didn't work for the file.

You need to tell the editor what programming language to associate with .egsinp files, since this is a custom extension. Telling it to pretend the file is Ruby tends to do a decent job of syntax highlighting and the comment types are right. In most general purpose editors, you can do this in the options. Spyder seems more limited.

**Reply by ftessier** [reply]
You can also try YAML syntax highlighting, or any of the many languages that use `#` for comments.
